api/tracking_service.py

```diff
@@ -0,0 +1,93 @@
+from fastapi import FastAPI, HTTPException
+from pydantic import BaseModel
+from utils import validate_client
+import redis
+import json
+
+app = FastAPI()
+# start connection to redis server
+r = redis.Redis(host="localhost", port=6379, db=0)
+
+# defines a new game in request body
+class Client(BaseModel):
+    user_id: int
+    game_id: int
+
+
+# Format
+# user_1 = {
+#   game_<id>: {
+#       data including number of guesses, boardState
+#   },
+#   game_<id>: {
+#       data including number of guesses, boardState
+#   }
+# }
+
+# redis-cli: HMGET <user_id> <game_id> should return the game info
+
+# Functionality for returning game status
+@app.get("/get_status")
```

Comment:
That was fast. Looks great

I think the only thing I'd consider is prefixing each of these like 

- `prefix-for-this-service/game/{game_id}/status` for getting the status
- `prefix/game/add` for add game
- `prefix/game/{game_id}/update` for updating the guesses. 

But I'm not sure if the `{game_id}` part is necessary in the path, especially in the update.  

just so the routing in traefik can use the one prefix `prefix-for-this-service` in the rule.  
or even just eliminate the `prefix-for-this-service` part and use `game` as the prefix. Since I don't think we have any other routes that use `game` as a pathprefix. 

I think that might eliminate the need for prefix stripping? But it's not really a functional requirement so whatever is fine.

Reply:
That makes sense actually, it would eliminate the need for prefix stripping, I just figured I'd keep the same structure as the checkings, validations, and statistics service. I admit it would appear more like a standard RESTful API with the prefixes in the route name but as you've said it's not a functional requirement so I think we should let it slide this one time 😃

Reply:
yeah i realized it would take too long to change all that up. since it's due in a few days. and wouldn't want to risk it not working. plus in project 5 it does end up using `/game`